tests/testthat/test-pointr.R

```diff
@@ -87,3 +87,23 @@ test_that("sharepoint_download errors on 404", {
 
   expect_false(file.exists(t))
 })
+
+test_that("sharepoint_download default save_path inherits file extension", {
+
+  ## TODO: this is a pretty poor test, but I don't know how to 'mock' download
```

Comment:
So the test failure is because it doesn't like the fact that I've used the word `TODO:` here.  I'm going to leave it here now for your review, but wanted to flag this is why the test is failing.

Reply:
The testing here is really weird because we don't have a test instance of sharepoint which we can send test queries to and using the imperial one would require having personal credentials stored somewhere which I don't want to do. 

The compromise in the testing is to mock out the the authentication steps - these are done with the `with_mock` stuff you have below. And then send actual requests to https://httpbin.org/ which is a service which allows us to test HTTP requests.

So I think for this test we want to use httpbin to download something with a filename `something.wahoo` and assert that the downloaded file has extension `.wahoo`. Then similar for the case with no file extension.

To do that I think if you change your download URL to something appropriate from httpbin.org that should support it - I think though atm it may have to the `anything` endpoint.  So line 104 would be `download <- sharepoint_download("https://httpbin.org", "/anything/something.wahoo")` and should get the correct file extension.

Reply:
Thanks for showing how to do this. I think addressed in fed9be8, but please check to make sure all tested that you would like.